Module#03/web-app/src/main/java/com/epam/esm/controller/CertificateController.java

```diff
@@ -0,0 +1,194 @@
+package com.epam.esm.controller;
+
+import com.epam.esm.assembler.CertificateAssembler;
+import com.epam.esm.assembler.TagAssembler;
+import com.epam.esm.dto.CertificateDto;
+import com.epam.esm.dto.TagDto;
+import com.epam.esm.service.CertificateService;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.web.PageableDefault;
+import org.springframework.hateoas.CollectionModel;
+import org.springframework.hateoas.EntityModel;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+import static org.springframework.http.HttpStatus.CREATED;
+import static org.springframework.http.HttpStatus.NO_CONTENT;
+
+/**
+ * Controller class for handling certificate-related operations.
+ * {@link RestController} to indicate that it is a Spring MVC controller,
+ * {@link RequestMapping} with a value of "/certificates" to map requests,
+ * and {@link CrossOrigin} with origins set to "*"
+ * and allowed headers set to "GET", "POST", "PUT", and "DELETE"
+ * to enable Cross-Origin Resource Sharing (CORS).
+ */
+@RestController
+@RequiredArgsConstructor
+@RequestMapping(value = "/certificates")
+@CrossOrigin(origins = "*", allowedHeaders = {"GET", "POST", "PUT", "DELETE"})
```

Comment:
if I am not wrong, those CrossOrigin is configured in WebMvcConfig class. So maybe this annotation is not needed here